src/App.tsx

```diff
@@ -17,18 +73,43 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onChangeQuery={setQuery}
+                selectedStatus={selectedStatus}
+                onSelectStatus={setSelectedStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length && (
+                <TodoList
+                  todos={visibleTodos}
+                  onSelectTodo={handleSelectTodo}
+                  selectedId={selectedTodo?.id}
+                />
+              )}
+              {!todos.length && <Loader />}
```

Comment:
What if there is no todos? You'll show an infinite loader which is not correct. You need another variable in state called for example `isLoading` or `isTodosLoading`. You set it to `true` before calling `getTodos` and set it to `false` after try/catch block in the function `loadTodos`